turbo/snapshotsync/freezeblocks/beacon_block_reader.go

```diff
@@ -45,48 +46,81 @@ func (r *beaconSnapshotReader) FrozenSlots() uint64 {
 }
 
 func (r *beaconSnapshotReader) ReadBlock(slot uint64) (*cltypes.SignedBeaconBlock, error) {
-	buf, err := r.RawBlockSSZ(slot)
-	if err != nil {
-		return nil, err
+	view := r.sn.View()
+	defer view.Close()
+
+	var buf []byte
+
+	seg, ok := view.BeaconBlocksSegment(slot)
+	if !ok {
+		return nil, nil
 	}
-	if buf == nil {
+
+	if seg.idxSlot == nil {
 		return nil, nil
 	}
+	blockOffset := seg.idxSlot.OrdinalLookup(slot - seg.idxSlot.BaseDataID())
```

Comment:
maybe need protect from `uint64` underflow by:
if slot < seg.idxSlot.BaseDataID() {
   return nil, nil
}
or return error in this case